OData/src/System.Web.OData/OData/EnableQueryAttribute.cs

```diff
@@ -534,16 +571,17 @@ private object ExecuteQuery(object response, HttpRequestMessage request, HttpAct
             Type elementClrType = GetElementType(response, actionDescriptor);
 
             IEdmModel model = GetModel(elementClrType, request, actionDescriptor);
-            if (model == null)
-            {
+            if (model == null) {
                 throw Error.InvalidOperation(SRResources.QueryGetModelMustNotReturnNull);
             }
 
-            ODataQueryContext queryContext = new ODataQueryContext(
-                model,
-                elementClrType,
-                request.ODataProperties().Path);
-            ODataQueryOptions queryOptions = new ODataQueryOptions(queryContext, request);
+            ODataQueryContext queryContext = CreateODataQueryContext(request, model, elementClrType);
+            ODataQueryOptions queryOptions = CreateODataQueryOptions(queryContext, request);
+
+            if (queryOptions.SelectExpand != null)
+            {
+                queryOptions.SelectExpand.LevelsMaxLiteralExpansionDepth = _validationSettings.MaxExpansionDepth;
```

Comment:
I don't undertand why this is related to SelectExpandQueryValidator. What you are mentioning does not allow us to create new instances of the ODataQueryContext which we need in order to override the function System.Web.OData.Query.ODataQueryOptions.ApplyTo(IQueryable query, System.Web.OData.Query.ODataQuerySettings querySettings).